tsl/src/compression/compression.c

```diff
@@ -2777,40 +2777,41 @@ fix_and_reorder_index_filters(Relation comp_chunk_rel, Relation index_rel,
 /*
  * A compressed chunk can have multiple indexes. For a given list
  * of columns in index_filters, find the matching index which has
- * all of the columns.
+ * the most columns based on index_filters and adjust the filters
+ * if necessary.
  * Return matching index if found else return NULL.
  *
- * Note: This method will not find the best matching index.
+ * Note: This method will find the best matching index based on
+ * number of filters it matches. If an index matches all the filters,
+ * it will be chosen. Otherwise, it will try to select the index
+ * which has most matches. If there are multiple indexes have
+ * the same number of matches, it will pick the first one it finds.
  * For example
  * for a given condition like "WHERE X = 10 AND Y = 8"
  * if there are multiple indexes like
  * 1. index (a,b,c,x)
  * 2. index (a,x,y)
  * 3. index (x)
- * 4. index (x,y)
- * In this case 2nd index is returned.
+ * In this case 2nd index is returned. If that one didn't exist,
+ * it would return the 1st index.
  */
 static Relation
-find_matching_index(Relation comp_chunk_rel, List *index_filters)
+find_matching_index(Relation comp_chunk_rel, List **index_filters, List **filters)
```

Comment:
Let's call it `heap_filters`? It's easy to confuse when you have one variable with prefix and one without.